### PR TITLE
Feature/sprint3 api development

### PR DESCRIPTION
**Pull request by FurkanKayar55**
Homecare - Brokerage Screen endpoint implementation
Supplier endpoint implementation
Homecare Brokerage stages endpoint implementation
Homecare Approval History endpoint implementation
Homecare - Approve Package Ordered endpoint implementation
Homecare - Approve Brokered Deal endpoint implementation
Seeding supplier and homecare Brokerage stages 
Add Migrations

### CODE REVIEW

File: LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using System.Threading.Tasks;
+
+namespace LBH.AdultSocialCare.Api.V1.Controllers.HomeCareApproveBrokered
+{
+    [Route("api/v1/home-care-approve-brokered")]
+    [Produces("application/json")]
+    [ApiController]
+    [ApiExplorerSettings(GroupName = "v1")]
+    [ApiVersion("1.0")]
+    public class HomeCareApproveBrokeredController : BaseController
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs:16]: To keep route hierarchy we can have child routes like these in the format `[Route("api/v1/home-care-packages/ {homeCarePackageId}/approve-brokered")]`

File: LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public HomeCareApproveBrokeredController(IGetHomeCareApproveBrokeredUseCase getHomeCareApproveBrokeredUseCase)
+        {
+            _getHomeCareApproveBrokeredUseCase = getHomeCareApproveBrokeredUseCase;
+        }
+
+        /// <summary>Gets the specified home care approve package brokered identifier.</summary>
+        /// <param name="homeCarePackageId">The home care package identifier.</param>
+        /// <returns>The home care approve brokered response.</returns>
+        [HttpGet]
+        [Route("{homeCarePackageId}")]
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs:29]: (Being a child route in home care hierarchy, `{homeCarePackageId}` can be moved to the top)

File: LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApprovePackage/HomeCareApprovePackageController.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+using LBH.AdultSocialCare.Api.V1.UseCase.HomeCareApprovePackageUseCase.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LBH.AdultSocialCare.Api.V1.Controllers.HomeCareApprovePackage
+{
+    [Route("api/v1/home-care-approve-package")]
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApprovePackage/HomeCareApprovePackageController.cs:11]: (Route hierarchy)

File: LBH.AdultSocialCare.Api/V1/Controllers/HomeCareBrokerage/HomeCareBrokerageController.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LBH.AdultSocialCare.Api.V1.Controllers.HomeCareBrokerage
+{
+    [Route("api/v1/home-care-brokerage")]
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Controllers/HomeCareBrokerage/HomeCareBrokerageController.cs:14]: (Route hierarchy). Maybe check other other controllers too

**Review by dunnudge — APPROVED** [review]
Solid work. Remember to run `dotnet format` to pass that formatting check

File: LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs (hunk 1 above)
Comment (FurkanKayar55) [reply LBH.AdultSocialCare.Api/V1/Controllers/HomeCareApproveBrokered/HomeCareApproveBrokeredController.cs:16]: Thank you make sense, I will.

**Review by petemade — APPROVED** [review]

**Pull request merged**